lib/api/apiUtils/object/checkObjectLock.js

```diff
@@ -0,0 +1,47 @@
+const { errors } = require('arsenal');
+
+function checkObjectLock(bucket, headers) {
+    const retentionObj = {};
+
+    if (!bucket.isObjectLockEnabled()) {
+        retentionObj.error = errors.InvalidRequest.customizeDescription(
+            'Bucket is missing ObjectLockConfiguration');
+        return retentionObj;
+    }
+    const headerMode = headers['x-amz-object-lock-mode'];
+    const headerRetainDate = headers['x-amz-object-lock-retain-until-date'];
+    const headerLegalHold = headers['x-amz-object-lock-legal-hold-status'];
```

Comment:
It might be worth moving these header strings into a const.js or something, they are used quite a bit and typos will be hard to find.